ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -331,8 +385,23 @@ int ldmsd_load_plugin(char* cfg_name, char *plugin_name,
 			errno, plugin_name);
 		goto err;
 	}
+	sprintf(&log_name[strlen(log_name)], "%s.%s", api->name, cfgobj->name);
+	ovis_log_t log = ovis_log_register(log_name, "Plugin log file");
+	if (!log)
+		ovis_log(NULL, OVIS_LWARN,
+			 "Error %d creating the log for the '%s' plugin.\n",
+			 errno, api->name);
+	if (api->type == LDMSD_PLUGIN_SAMPLER)
+		sampler->log = log;
+	else
+		store->log = log;
+	free((char *)libpath);
 	return 0;
  err:
```

Comment:
We need to clean up after load_plugin() here: free libpath and dlclose() the handle returned by dlopen().